src/core/flow.js

```diff
@@ -352,9 +360,18 @@ export function FlowWithPhases({
 
   // Helper to perform start-of-phase initialization.
   const startPhase = function(state, config) {
-    const G = config.onPhaseBegin(state.G, ctx);
-
     const ctx = { ...state.ctx };
+    if (config.secondsPerPhase) {
+      ctx.phaseElapsedTime = 0;
+      const interval = setInterval(() => {
+        if (++ctx.phaseElapsedTime >= config.secondsPerPhase) {
+          clearInterval(interval);
+          state = endPhaseEvent(state);
```

Comment:
The only clean way to do this would be to dispatch the `endPhase` event to the store directly. Right now, I don't think this change would be propagated to clients correctly.

I would suggest that you revert the most recent commit about storing elapsed time to keep things simple (I don't think we need to support that; it's easy enough to have a client side timer to show the remaining time).